GraknClient.java

```diff
@@ -617,9 +625,55 @@ public boolean isClosed() {
             }
         }
 
+        private static class TransactionIterator extends AbstractIterator<SessionProto.Transaction.Iter.Res> implements Transceiver.ResponseReceiver {
+            private final BlockingQueue<Transceiver.Response> queue = new LinkedBlockingQueue<>();
+
+            @Override
+            public boolean onResponse(Transceiver.Response response) {
+                queue.add(response);
+                if (response.type() == OK) {
+                    SessionProto.Transaction.Iter.Res iterRes = response.ok().getIterRes();
+                    switch (iterRes.getResCase()) {
+                        case DONE:
+                        case ITERATORID:
+                            return false;
+                        default:
+                            return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            protected SessionProto.Transaction.Iter.Res computeNext() {
+                Transceiver.Response response;
+                try {
+                    response = queue.take();
```

Comment:
What happens when the queue is empty (but more responses is yet to come) and `queue.take()` is called?

Reply:
`queue.take()` is the blocking form (as opposed to `.poll()` which does not block), so this will wait for the next result.

Reply:
👍